blockstore/splitstore/markset_bloom.go

```diff
@@ -64,12 +67,20 @@ func (s *BloomMarkSet) saltedKey(cid cid.Cid) []byte {
 }
 
 func (s *BloomMarkSet) Mark(cid cid.Cid) error {
-	s.bf.Add(s.saltedKey(cid))
+	if s.ts {
+		s.bf.AddTS(s.saltedKey(cid))
+	} else {
+		s.bf.Add(s.saltedKey(cid))
+	}
+
 	return nil
 }
 
 func (s *BloomMarkSet) Has(cid cid.Cid) (bool, error) {
-	return s.bf.Has(s.saltedKey(cid)), nil
+	if s.ts {
+		return s.bf.HasTS(s.saltedKey(cid)), nil
+	}
+	return s.bf.HasTS(s.saltedKey(cid)), nil
 }
 
 func (s *BloomMarkSet) Close() error {
```

Comment:
Probably want to make the bloom nil (optionally under the lock), so it can be reclaimed.

Reply:
sure, will do.

Reply:
done.

Reply:
ditto here, actually unsafe so I have reverted.

Reply:
readded with explicit nil checks, see above.